app/server.js

```diff
@@ -0,0 +1,196 @@
+  .addField('indeed', 'My depth is immaterial to this conversation.')
+  .addField('so good|feels good', 'Feels good man !')
+  .addField("n'est-ce pas ?", "You need the ol' racist uncle card ? Don't move !");
+
+function duplicatedLinkDetection(message) {
+    const hashedUrl = SHA256(message.content).toString();
+
+    redisClient.get(hashedUrl, (err, reply) => {
+        if (!reply) {
+            // link is fresh or at least 12 hours old
+            // save the hash and links meta in redis for 12 hours
+            const data = {
+                user: message.author.username,
+                channel: message.channel.name,
+                date: message.createdAt,
+            };
+            redisClient.set(hashedUrl, JSON.stringify(data), 'EX', TWELVE_HOURS);
+        }
+
+        if (reply && message.deletable) {
+            // IT'S A REPOST :SCREAM:
+            // delete the message, send the repost PNG
+            // and blame the user with a custom blame message
+            const data = JSON.parse(reply);
+            const date = moment(new Date(data.date));
+            const blameMessage = `${data.user} posted this in #${data.channel} on ${date.fromNow()}`;
+            message.channel.send(blameMessage, {
+                file: `${PATH_TO_STATIC_FOLDER}/no_repost.jpg`,
+            });
+            message.delete();
+        }
+    });
+}
+
+function invokeDenis(message) {
+    message.channel.send('', {file: `${PATH_TO_STATIC_FOLDER}/ah.png`});
+}
+
+function invokeJML(message) {
+    message.channel.send('', {file: `${PATH_TO_STATIC_FOLDER}/nestcepas.gif`});
+}
+
+function invokeTheDonald(message) {
+// will select randomly between fakenews_1.jpg and fakenews_2.jpg
+    const image = ['fakenews_1', 'fakenews_2'][Math.round(Math.random())];
+    message.channel.send('', {file: `${PATH_TO_STATIC_FOLDER}/${image}.jpg`});
+}
+
+function invokeJupiter(message) {
+    message.channel.send('', {file: `${PATH_TO_STATIC_FOLDER}/projet.gif`});
+}
+
+function getVancouverTime(message) {
+    const vancouverDateTime = moment()
+        .tz('America/Vancouver')
+        .locale('fr')
+        .format('LT');
+    message.channel.send(`:flag_ca: ${vancouverDateTime} :maple_leaf:`);
+}
+
+function avoirHeureParis(message) {
+    const parisDateTime = moment()
+        .tz('Europe/Paris')
+        .locale('fr')
+        .format('LT');
+    message.channel.send(`:flag_fr: ${parisDateTime} :french_bread:`);
+}
+
+function getHelpCommand(message) {
+    message.channel.send(helpMessage);
+}
+
+function invokeTealc(message) {
+    message.channel.send('', {file: `${PATH_TO_STATIC_FOLDER}/indeed.gif`});
+}
+
+function invokeGoodMan(message) {
+    message.channel.send('', {file: `${PATH_TO_STATIC_FOLDER}/so_good.png`});
+}
+
+/*
+ * REAL MAGIC HAPPENS
+ */
+client.on('message', (message) => {
+  // do not need to react to somebody not alive, right ?
+  if (message.author.bot) {
```

Comment:
```javascript
if (message.author.bot) return false;
```

Reply:
I really dislike this kind of writing 🤮

Reply:
Why ?

Reply:
Because it is a very inconsistent way of coding style. When reading you'll come across 

```
if (condition) then
```

and

```
if (condition) {
  then
}
```

which when you read is easy to miss something.

Reply:
That's why you need `Prettier` my friend :)